var/spack/repos/builtin/packages/flux-core/package.py

```diff
@@ -125,6 +128,10 @@ class FluxCore(AutotoolsPackage):
     # Patch 0.27-0.30 for build errors when czmq built with "draft APIs":
     patch("0001-build-fix-build-errors-with-side-installed-0MQ.patch", when="@0.27.0:0.30.0")
 
+    def patch(self):
+        with when("^python@3.12:"):
+            filter_file("^import sys", "import sys;import setuptools", "configure")
```

Comment:
This probably needs a flux-core version guard now, since we added the appropriate changes in a newer release.